src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec.py

```diff
@@ -0,0 +1,137 @@
+        raise MalformedBNFError(self.line)
+
+    def _parseStandardRule(self, standard: Match[str]) -> StandardSyntacticRule:
+        self.lhs, self.rhs = standard["lhs"], standard["rhs"]
+        return StandardSyntacticRule(self.line, self._parseLeft(), self._parseRight())
+
+    def _parseLeft(self) -> LhsNonTerminal:
+        match = self._matchLeft()
+        if not match:
+            raise MalformedLHSError(self.line)
+        return LhsNonTerminal(match["nonTerminal"], match["altName"])
+
+    def _parseRight(self) -> List[Symbol]:
+        if self.rhs is None:
+            return []
+        return [
+            self._parseSymbol(symbol)
+            for symbol in self.rhs.split()
+            if symbol and not symbol.startswith("#")
+        ]
+
+    def _parseSymbol(self, symbol: str) -> Symbol:
+        capturing = re.match(r"<(?P<name>\S*)>(?P<altName>\S+)?", symbol)
+        return (
+            self._parseCapturing(capturing["name"], capturing["altName"])
+            if capturing
+            else Terminal(symbol)
+        )
+
+    def _parseCapturing(self, name: str, altName: str) -> CapturingSymbol:
+        terminal = re.match(r"[A-Z_]+", name)
+        altName = altName.strip(":") if altName is not None else altName
+        return (
+            CapturingTerminal(name, altName)
+            if terminal
+            else RhsNonTerminal(name, altName)
+        )
+
+    def _buildMatches(
+        self,
+    ) -> tuple[Match[str] | None, Match[str] | None, Match[str] | None]:
+        lineStr = self.line.string.split("#")[0].strip()
+        standard = re.match(
+            r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*::=(?:\s(?P<rhs>.*))?", lineStr
+        )
+        separated = re.match(
+            r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)(?P<separator>\+.*)",
+            lineStr,
+        )
+        repeating = re.match(
+            r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)", lineStr
+        )
+        return standard, separated, repeating
+
+    def _parseSeparatedRule(self, separated: Match[str]) -> RepeatingSyntacticRule:
+        self.lhs, self.rhs, self.separator = (
+            separated["lhs"],
+            separated["rhs"],
+            separated["separator"].strip("+").strip(" "),
+        )
+        return self._parseRepeating()
+
+    def _parseRepeatingRule(self, repeating: Match[str]) -> RepeatingSyntacticRule:
+        self.lhs, self.rhs = repeating["lhs"], repeating["rhs"]
+        return self._parseRepeating()
+
+    def _parseRepeating(self) -> RepeatingSyntacticRule:
+        return RepeatingSyntacticRule(
+            self.line,
+            self._parseLeft(),
+            self._parseRight(),
+            Terminal(self.separator) if self.separator else None,
+        )
+
+    def _matchLeft(self) -> Match[str] | None:
+        return re.match(r"<(?P<nonTerminal>\S*)>(?::(?P<altName>\S+))?\s*", self.lhs)
+
+    def _isSkippable(self) -> bool:
```

Comment:
_isSkippable is used beyond this class. So either restructure so it isn't or name it isSkippable (no underscore).

Reply:
This is only used in a method with a negation (a "not" in front). In such cases, I like to try to rename the method so that the negation isn't needed.

What about isSyntacticRule()?

You'll need to move the "not" into the method. But then that makes the call cleaner and more descriptive... sic:

    if not parser._isSkippable():
        self.spec.append(parser.parseSyntacticRule())
vs

    if parser.isSyntacticRule():
        self.spec.append(parser.parseSyntacticRule())